ui/packages/consul-peerings/app/controllers/dc/peers/edit/exported.js

```diff
@@ -0,0 +1,17 @@
+import Controller from "@ember/controller";
+import { tracked } from "@glimmer/tracking";
+import { action } from "@ember/object";
+
+export default class PeersEditExportedController extends Controller {
+  queryParams = {
```

Comment:
I remember John saying that the app didn't use query params as prescribed by Ember (declaring them in the controller like this) I don't really remember the reasoning. Happy to see this approach, but figured it would be prudent to ask if there could be any issues with how it's working elsewhere. I imagine not an issue.

Reply:
It's not an issue to declare the query parameter in the controller like you would do in a regular app. Actually, I had to declare it like this because without the query parameter setup on the controller, the URL wouldn't update when changing `search`. So, why does this work for `peers.edit.imported` and not `peers.edit.exported`?  I dug a little bit to find out why query parameter work with the custom DSL but weren't working here when declaring the controller file:

It turns out that declaring the query parameters on the route is good enough when you use the `mut`-helper to create a two-way data-binding for the search property. We do in the regular `SearchBar`-setup, [that is used across the rest of the application](https://github.com/hashicorp/consul/pull/14947/files#diff-f19b6467ddb33d4df2c09ab9007d19a886adcd842e1e24d6811a908b2c009fb7R62), but we aren't doing that in the exported-template as I wanted to stick to data-down-actions-up as much as possible for new code.

__TLDR;__: We actually need to declare the query-param in the controller, when we want to use a tracked property that should update a query-parameter in the URL.